### ISSUE TITLE
preview window won't disappear after sorting

### ISSUE THREAD

**Issue by jasonPaddy** [sjarvela/kloudspeaker#69]
Hi Sam,

Another minor issue.
Symptom:
Click on any file, preview/action window pop up, click any sorting, pop up window stuck
only fix is to refresh the page

![screen shot 2015-09-30 at 11 10 01 am](https://cloud.githubusercontent.com/assets/12420731/10182106/209d6e7a-6710-11e5-9fe9-7add569e7c16.png)

**Comment by sjarvela** [comment]
Popovers, tooltips etc are hidden whenever view is clicked, but it turns out that when sort action is triggered, it recreates the content _before_ it hides the popup. Problem here is that the element which had the popover, does not exist anymore, and popover cannot close itself.

I made a quick fix using invokeLater on sort, but I have to investigate if this happens on other places, and also if there is a way to hide the popup even if the parent element has been removed.
